plugins/auto-sizes/hooks.php

```diff
@@ -139,79 +139,147 @@ function auto_sizes_get_width( string $layout_width, int $image_width ): string
 }
 
 /**
- * Filter the sizes attribute for images to improve the default calculation.
+ * Primes attachment into the cache with a single database query.
  *
- * @since 1.1.0
+ * @since n.e.x.t
  *
- * @param string                                                   $content      The block content about to be rendered.
- * @param array{ attrs?: array{ align?: string, width?: string } } $parsed_block The parsed block.
- * @return string The updated block content.
+ * @param string $content The HTML content.
+ * @param string $context Optional. Additional context to pass to the filters.
+ *                        Defaults to `current_filter()` when not set.
+ * @return string The HTML content.
  */
-function auto_sizes_filter_image_tag( string $content, array $parsed_block ): string {
-	$processor = new WP_HTML_Tag_Processor( $content );
-	$has_image = $processor->next_tag( array( 'tag_name' => 'img' ) );
+function auto_sizes_prime_attachment_caches( string $content, string $context = null ): string {
+	if ( null === $context ) {
+		$context = current_filter();
+	}
 
-	// Only update the markup if an image is found.
-	if ( $has_image ) {
-		$processor->set_attribute( 'data-needs-sizes-update', true );
-		if ( isset( $parsed_block['attrs']['align'] ) ) {
-			$processor->set_attribute( 'data-align', $parsed_block['attrs']['align'] );
-		}
+	if ( false === preg_match_all( '/<img\s[^>]+>/', $content, $matches, PREG_SET_ORDER ) ) {
```

Comment:
Wouldn't this be better using HTML Tag Processor?

Reply:
Done in https://github.com/WordPress/performance/pull/1625/commits/b87adcfa0481c964ea61ab0d5053a2ada6957369